testkit/src/lib.rs

```diff
@@ -16,75 +16,57 @@
-//! #[macro_use]
-//! extern crate exonum_derive;
-//! #[macro_use]
-//! extern crate exonum_testkit;
-//! extern crate serde_json;
-//! #[macro_use] extern crate serde_derive;
-//! extern crate failure;
-//!
-//! use serde_json::Value;
-//! use std::borrow::Cow;
-//!
-//! use exonum::api::node::public::explorer::{BlocksQuery, BlocksRange, TransactionQuery};
-//! use exonum::blockchain::{Block, Schema, Service, Transaction, TransactionContext,
-//! TransactionSet, ExecutionResult};
-//! use exonum::crypto::{gen_keypair, Hash, PublicKey, SecretKey, CryptoHash};
-//! use exonum::explorer::TransactionInfo;
-//! use exonum::helpers::Height;
-//! use exonum::messages::{Signed, AnyTx, Message};
-//! use exonum_merkledb::{Snapshot, Fork};
-//! use exonum_testkit::{ApiKind, TestKitBuilder};
+//! ```
+//! use exonum::{
+//!     runtime::{InstanceDescriptor, rust::{Transaction, TransactionContext, Service}},
+//!     blockchain::{Block, Schema, ExecutionError, InstanceCollection},
+//!     crypto::{gen_keypair, Hash},
+//!     explorer::TransactionInfo,
+//!     helpers::Height,
+//!     api::node::public::explorer::{BlocksQuery, BlocksRange, TransactionQuery},
+//! };
+//! use serde_derive::{Serialize, Deserialize};
+//! use exonum_derive::{exonum_service, ServiceFactory, ProtobufConvert};
+//! use exonum_merkledb::{ObjectHash, Snapshot};
+//! use exonum_testkit::{txvec, ApiKind, TestKitBuilder};
 //!
 //! // Simple service implementation.
 //!
-//! const SERVICE_ID: u16 = 1;
+//! const SERVICE_ID: u32 = 1;
 //!
 //! #[derive(Debug, Clone, Serialize, Deserialize, ProtobufConvert)]
 //! #[exonum(pb = "exonum_testkit::proto::examples::TxTimestamp")]
-//! struct TxTimestamp {
+//! pub struct TxTimestamp {
 //!     message: String,
 //! }
 //!
-//! #[derive(Debug, Clone, Serialize, Deserialize, TransactionSet)]
-//! enum TimestampingTransactions {
-//!     TxTimestamp(TxTimestamp),
-//! }
-//!
-//! impl TxTimestamp {
-//!    fn sign(author: &PublicKey, msg: &str, key: &SecretKey) -> Verified<AnyTx> {
-//!        let tx = TxTimestamp{ message: msg.to_owned() };
-//!        Message::sign_transaction(tx, SERVICE_ID, *author, key)
-//!    }
-//! }
-//!
+//! #[derive(Clone, Default, Debug, ServiceFactory)]
+//! #[exonum(
+//!     artifact_name = "timestamping",
+//!     artifact_version = "1.0.0",
+//!     proto_sources = "exonum_testkit::proto",
+//!     implements("TimestampingInterface")
+//! )]
 //! struct TimestampingService;
 //!
-//! impl Transaction for TxTimestamp {
-//!     fn execute(&self, _context: TransactionContext) -> ExecutionResult {
-//!         Ok(())
-//!     }
-//! }
-//!
 //! impl Service for TimestampingService {
-//!     fn service_name(&self) -> &str {
-//!         "timestamping"
-//!     }
+//!     fn state_hash(&self, _: InstanceDescriptor, _: &dyn Snapshot) -> Vec<Hash> { vec![] }
+//! }
 //!
-//!     fn state_hash(&self, _: &Snapshot) -> Vec<Hash> {
-//!         Vec::new()
+//! impl From<TimestampingService> for InstanceCollection {
```

Comment:
I'm not sure if this is logically correct to implement such a conversion, especially in the example.
And another thought: since we expect that user will test already implemeted services, maybe hide service implementation from docs (if we'll rename service to ExampleService and tx to ExampleTx, it will be self-explanatory enough IMO)?

Reply:
Removed these From impls, renamed My* to Example*.

> hide service implementation from docs

They seem to be already hidden (at least in the doc-tests that this PR touches), except for the top-level crate example that is supposed (as I see it) to be a minimal but complete example.